### PR TITLE
Add graalvm weblog variant

### PR DESCRIPTION
**Pull request by robertomonteromiguel**
## Description

New weblog variant in order to test graalvm support

## Check list

Your PR is not ready to be reviewed? Please save it as a draft :pray:

- [ ] Follows the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
- [ ] CI is passing
- [ ] There is at least one approval from code owners

Yes to all? Feel free to merge it whenever you want :heart:

### CODE REVIEW

File: utils/build/docker/java/spring-boot-3-native/pom.xml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+                        <id>build-native</id>
+                        <goals>
+                            <goal>build</goal>
+                        </goals>
+                        <phase>package</phase>
+                    </execution>
+                </executions>
+                <configuration>
+                     <buildArgs>
+                        <buildArg>-H:+ReportExceptionStackTraces</buildArg>
```
Comment (PerfectSlayer) [inline utils/build/docker/java/spring-boot-3-native/pom.xml:82]: Same question

File: utils/build/docker/java/spring-boot-native.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+COPY ./utils/build/docker/java/spring-boot/src ./src
+RUN mv ./src/main/resources/application-native.properties ./src/main/resources/application.properties
+
+# Copy tracer
+COPY --from=agent /dd-tracer/dd-java-agent.jar .
+
+# Build native application
+RUN /opt/apache-maven-3.8.6/bin/mvn package -P spring-native
+
+FROM adoptopenjdk:11-jre-hotspot
```
Comment (PerfectSlayer) [inline utils/build/docker/java/spring-boot-native.Dockerfile:32]: Is hotspot required? What about using only `ubuntu` to ensure Java is not used / required?

File: utils/build/docker/java/spring-boot-3-native.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+RUN /opt/apache-maven-3.8.6/bin/mvn package -P native
+
+FROM ubuntu
+
+WORKDIR /app
+COPY --from=agent /binaries/SYSTEM_TESTS_LIBRARY_VERSION SYSTEM_TESTS_LIBRARY_VERSION
+COPY --from=agent /binaries/SYSTEM_TESTS_LIBDDWAF_VERSION SYSTEM_TESTS_LIBDDWAF_VERSION
+COPY --from=agent /binaries/SYSTEM_TESTS_APPSEC_EVENT_RULES_VERSION SYSTEM_TESTS_APPSEC_EVENT_RULES_VERSION
+COPY --from=build /app/target/myproject .
+COPY --from=build /opt/graalvm-ce-java17-22.3.0/ /opt/graalvm-ce-java17-22.3.0/
```
Comment (PerfectSlayer) [inline utils/build/docker/java/spring-boot-3-native.Dockerfile:38]: Is graalvm mandatory to run?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -73,6 +71,10 @@ jobs:
           weblog: spring-boot-openliberty
         - library: java
           weblog: spring-boot-undertow
+        - library: java
+          weblog: spring-boot-native
```
Comment (cbeauchesne) [inline .github/workflows/ci.yml:75]: Could you keep the `spring-boot` name? it is used in AppSe feature matrix

**Review by cbeauchesne — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 4 above)
Comment (PerfectSlayer) [reply .github/workflows/ci.yml:75]: This name defines the docker image to build the weblog variant.
How can we keep the same name?

Comment (robertomonteromiguel) [reply .github/workflows/ci.yml:75]: It's the same case as "spring-boot-undertow" and "spring-boot-openliberty", they are build profiles defined in spring-boot app. I think we can not  keep the spring-boot name....

File: tests/test_miscs.py [hunk 5]
```diff
@@ -2,7 +2,7 @@
 # This product includes software developed at Datadog (https://www.datadoghq.com/).
 # Copyright 2022 Datadog, Inc.
 
-from utils import weblog, interfaces, released, irrelevant
+from utils import weblog, interfaces, released, irrelevant, bug
```
[the remark this answers is not in the document]
Comment (robertomonteromiguel) [reply tests/test_miscs.py:5]: ok. Removed!

File: utils/build/docker/java/spring-boot/pom.xml [hunk 6]
```diff
@@ -282,6 +282,91 @@
...
+                                <id>build-native</id>
+                                <goals>
+                                    <goal>build</goal>
+                                </goals>
+                                <phase>package</phase>
+                            </execution>
+                        </executions>
+                        <configuration>
+                            <buildArgs>
+                                <buildArg>-H:+ReportExceptionStackTraces</buildArg>
```
[the remark this answers is not in the document]
Comment (robertomonteromiguel) [reply utils/build/docker/java/spring-boot/pom.xml:307]: ok. Removed!

File: utils/build/docker/java/spring-boot/pom.xml [hunk 7]
```diff
@@ -325,6 +410,22 @@
         <protobuf.version>3.16.3</protobuf.version>
         <grpc.version>1.31.0</grpc.version>
         <packaging.type>jar</packaging.type>
+		<agent.path>/app/dd-java-agent.jar</agent.path>
+		<!-- Enable for Mac M1 workaround-->
+		<!-- <os.detected.classifier>osx-x86_64</os.detected.classifier> -->
```
[the remark this answers is not in the document]
Comment (robertomonteromiguel) [reply utils/build/docker/java/spring-boot/pom.xml:415]: Ok. I created a new spring-native-local profile

File: utils/build/docker/java/spring-boot-3-native/pom.xml (hunk 1 above)
Comment (robertomonteromiguel) [reply utils/build/docker/java/spring-boot-3-native/pom.xml:82]: ok. Removed!

File: utils/build/docker/java/spring-boot-3-native.Dockerfile (hunk 3 above)
Comment (robertomonteromiguel) [reply utils/build/docker/java/spring-boot-3-native.Dockerfile:38]: I don't known why, but if I remove it:

`Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'jsonMixinModule': I/O failure during classpath scanning
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1754) ~[myproject:6.0.0-M6]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:599) ~[myproject:6.0.0-M6]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:521) ~[myproject:6.0.0-M6]
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:326) ~[myproject:6.0.0-M6]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[myproject:6.0.0-M6]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:324) ~[myproject:6.0.0-M6]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:200) ~[myproject:6.0.0-M6]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.getBeansOfType(DefaultListableBeanFactory.java:662) ~[myproject:6.0.0-M6]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.getBeansOfType(DefaultListableBeanFactory.java:650) ~[myproject:6.0.0-M6]
	at org.springframework.context.support.AbstractApplicationContext.getBeansOfType(AbstractApplicationContext.java:1287) ~[myproject:6.0.0-M6]
	at org.springframework.beans.factory.BeanFactoryUtils.beansOfTypeIncludingAncestors(BeanFactoryUtils.java:321) ~[na:na]
	at org.springframework.boot.autoconfigure.jackson.JacksonAutoConfiguration$Jackson2ObjectMapperBuilderCustomizerConfiguration$StandardJackson2ObjectMapperBuilderCustomizer.getBeans(JacksonAutoConfiguration.java:331) ~[myproject:na]
	at org.springframework.boot.autoconfigure.jackson.JacksonAutoConfiguration$Jackson2ObjectMapperBuilderCustomizerConfiguration$StandardJackson2ObjectMapperBuilderCustomizer.configureModules(JacksonAutoConfiguration.java:295) ~[myproject:na]
	at org.springframework.boot.autoconfigure.jackson.JacksonAutoConfiguration$Jackson2ObjectMapperBuilderCustomizerConfiguration$StandardJackson2ObjectMapperBuilderCustomizer.customize(JacksonAutoConfiguration.java:200) ~[myproject:na]
	at org.springframework.boot.autoconfigure.jackson.JacksonAutoConfiguration$JacksonObjectMapperBuilderConfiguration.customize(JacksonAutoConfiguration.java:148) ~[myproject:na]
	at org.springframework.boot.autoconfigure.jackson.JacksonAutoConfiguration$JacksonObjectMapperBuilderConfiguration.jacksonObjectMapperBuilder(JacksonAutoConfiguration.java:141) ~[myproject:na]
	at org.springframework.boot.autoconfigure.jackson.JacksonAutoConfiguration_JacksonObjectMapperBuilderConfiguration__BeanDefinitions.lambda$getJacksonObjectMapperBuilderInstanceSupplier$0(JacksonAutoConfiguration_JacksonObjectMapperBuilderConfiguration__BeanDefinitions.java:31) ~[na:na]
	at org.springframework.util.function.ThrowingBiFunction.apply(ThrowingBiFunction.java:68) ~[myproject:6.0.0-M6]
	at org.springframework.util.function.ThrowingBiFunction.apply(ThrowingBiFunction.java:54) ~[myproject:6.0.0-M6]
	at org.springframework.beans.factory.aot.BeanInstanceSupplier.get(BeanInstanceSupplier.java:210) ~[na:na]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.obtainInstanceFromSupplier(AbstractAutowireCapableBeanFactory.java:1225) ~[myproject:6.0.0-M6]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.obtainFromSupplier(AbstractAutowireCapableBeanFactory.java:1210) ~[myproject:6.0.0-M6]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1157) ~[myproject:6.0.0-M6]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:561) ~[myproject:6.0.0-M6]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:521) ~[myproject:6.0.0-M6]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:344) ~[myproject:6.0.0-M6]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:200) ~[myproject:6.0.0-M6]
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:254) ~[myproject:6.0.0-M6]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1375) ~[myproject:6.0.0-M6]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1295) ~[myproject:6.0.0-M6]
	at org.springframework.beans.factory.aot.BeanInstanceSupplier.resolveArgument(BeanInstanceSupplier.java:332) ~[na:na]
	... 80 common frames omitted
Caused by: org.springframework.beans.factory.BeanDefinitionStoreException: I/O failure during classpath scanning
	at org.springframework.context.annotation.ClassPathScanningCandidateComponentProvider.scanCandidateComponents(ClassPathScanningCandidateComponentProvider.java:468) ~[myproject:6.0.0-M6]
	at org.springframework.context.annotation.ClassPathScanningCandidateComponentProvider.findCandidateComponents(ClassPathScanningCandidateComponentProvider.java:316) ~[myproject:6.0.0-M6]
	at org.springframework.boot.jackson.JsonMixinModule.afterPropertiesSet(JsonMixinModule.java:73) ~[myproject:3.0.0-M5]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1800) ~[myproject:6.0.0-M6]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1750) ~[myproject:6.0.0-M6]
	... 110 common frames omitted
Caused by: java.nio.file.NoSuchFileException: /opt/graalvm-ce-java17-22.3.0/lib/svm/builder/pointsto.jar
	at java.base@17.0.5/sun.nio.fs.UnixFileAttributeViews$Basic.readAttributes(UnixFileAttributeViews.java:55) ~[myproject:na]
	at java.base@17.0.5/sun.nio.fs.UnixFileSystemProvider.readAttributes(UnixFileSystemProvider.java:148) ~[myproject:na]
	at java.base@17.0.5/sun.nio.fs.LinuxFileSystemProvider.readAttributes(LinuxFileSystemProvider.java:99) ~[na:na]
	at java.base@17.0.5/java.nio.file.Files.readAttributes(Files.java:1851) ~[na:na]
	at java.base@17.0.5/java.util.zip.ZipFile$Source.get(ZipFile.java:1264) ~[na:na]
	at java.base@17.0.5/java.util.zip.ZipFile$CleanableResource.<init>(ZipFile.java:709) ~[na:na]
	at java.base@17.0.5/java.util.zip.ZipFile.<init>(ZipFile.java:243) ~[myproject:na]
	at java.base@17.0.5/java.util.zip.ZipFile.<init>(ZipFile.java:172) ~[myproject:na]
	at java.base@17.0.5/java.util.jar.JarFile.<init>(JarFile.java:347) ~[na:na]
	at java.base@17.0.5/jdk.internal.module.ModuleReferences$JarModuleReader.newJarFile(ModuleReferences.java:230) ~[na:na]
	at java.base@17.0.5/jdk.internal.module.ModuleReferences$JarModuleReader.<init>(ModuleReferences.java:237) ~[na:na]
	at java.base@17.0.5/jdk.internal.module.ModuleReferences.lambda$newJarModule$0(ModuleReferences.java:94) ~[na:na]
	at java.base@17.0.5/jdk.internal.module.ModuleReferenceImpl.open(ModuleReferenceImpl.java:93) ~[na:na]
	at org.springframework.core.io.support.PathMatchingResourcePatternResolver.lambda$findAllModulePathResources$4(PathMatchingResourcePatternResolver.java:901) ~[myproject:6.0.0-M6]
	at java.base@17.0.5/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183) ~[na:na]
	at java.base@17.0.5/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179) ~[na:na]
	at java.base@17.0.5/java.util.Iterator.forEachRemaining(Iterator.java:133) ~[myproject:na]
	at java.base@17.0.5/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1845) ~[na:na]
	at java.base@17.0.5/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[myproject:na]
	at java.base@17.0.5/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[myproject:na]
	at java.base@17.0.5/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150) ~[myproject:na]
	at java.base@17.0.5/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173) ~[na:na]
	at java.base@17.0.5/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[myproject:na]
	at java.base@17.0.5/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596) ~[myproject:na]
	at org.springframework.core.io.support.PathMatchingResourcePatternResolver.findAllModulePathResources(PathMatchingResourcePatternResolver.java:899) ~[myproject:6.0.0-M6]
	at org.springframework.core.io.support.PathMatchingResourcePatternResolver.getResources(PathMatchingResourcePatternResolver.java:318) ~[myproject:6.0.0-M6]
	at org.springframework.context.support.AbstractApplicationContext.getResources(AbstractApplicationContext.java:1408) ~[myproject:6.0.0-M6]
	at org.springframework.context.support.GenericApplicationContext.getResources(GenericApplicationContext.java:265) ~[myproject:6.0.0-M6]
	at org.springframework.context.annotation.ClassPathScanningCandidateComponentProvider.scanCandidateComponents(ClassPathScanningCandidateComponentProvider.java:421) ~[myproject:6.0.0-M6]
	... 114 common frames omitted`

Comment (PerfectSlayer) [reply utils/build/docker/java/spring-boot-3-native.Dockerfile:38]: I spent few hours to fix it.
I will make a PR to merge the fix on our branch.

**Review by PerfectSlayer — APPROVED** [review]

**Review by cbeauchesne — COMMENTED** [review]
Why did you ran only scenario of the first set ?

**Comment by robertomonteromiguel** [comment]
> Why did you ran only scenario of the first set ?

Because, we have only support for tracing with GraalVM (no appsec, debug, profiling....)

**Pull request merged**